stdpopsim/genomes.py

```diff
@@ -309,89 +316,109 @@ def species_contig(
     def length(self):
         return self.recombination_map.sequence_length
 
-    @property
     def all_intervals_array(self):
         """
-        Returns an (n, 3) numpy array with the intervals across all genomic
-        element types in the form (left, right, type).
+        Returns a dictionary with two things: "intervals" is the (n, 2) array
+        of all intervals across all DFEs (sorted and nonoverlapping), and
+        "dfes" is a length-n array of integer labels, so that the DFE that
+        applies to ``intervals[i]`` is ``dfe_list[dfes[i]]``.
         """
-        all_intervals = np.concatenate(
-            [
-                np.column_stack((g[:, :2], np.full((g.shape[0]), i)))
-                for i, g in enumerate(self.interval_list)
-            ],
+        intervals = np.concatenate(
+            [g[:, :2] for g in self.interval_list],
+            axis=0,
+        )
+        labels = np.concatenate(
+            [np.full((g.shape[0],), i) for i, g in enumerate(self.interval_list)],
             axis=0,
         )
-        all_intervals = stdpopsim.utils.build_intervals_array(all_intervals)
-        return all_intervals
+        intervals = stdpopsim.utils.build_intervals_array(intervals)
+        return {"intervals": intervals, "dfes": labels}
 
     def clear_DFEs(self):
```

Comment:
Is this used outside the tests? Maybe we can remove this method (or at least make it private)?

Reply:
i think people might use it in workflows?

Reply:
Since contigs come with a neutral DFE covering everything, it seems like we *ought* to provide a method to remove it... but, I'm not totally sure what the use case is (only simulating selected mutations I guess)?